android/src/main/java/com/rncamerakit/CKCamera.kt

```diff
@@ -307,9 +308,42 @@ class CKCamera(context: ThemedReactContext) : FrameLayout(context), LifecycleObs
         val useCases = mutableListOf(preview, imageCapture)
 
         if (scanBarcode) {
-            val analyzer = QRCodeAnalyzer { barcodes ->
-                if (barcodes.isNotEmpty()) {
-                    onBarcodeRead(barcodes)
+            val analyzer = QRCodeAnalyzer { results ->
+                if (results.isNotEmpty()) {
+                    val filteredResults = if (barcodeFrame !== null) {
+                        results.filter { (barcode, imageSize) ->
+                            val barcodeBoundingBox = barcode.boundingBox
+                            if (barcodeBoundingBox === null) {
+                                return@filter false
+                            }
+
+                            // val scaleX = viewFinder.width.toFloat() / imageSize.width
+                            // val scaleY = viewFinder.height.toFloat() / imageSize.height
+
+                            // Calculate scaling factors (image is always rotated by 90 degrees)
+                            val scaleX = viewFinder.width.toFloat() / imageSize.height
+                            val scaleY = viewFinder.height.toFloat() / imageSize.width
+                            val scaledBarcodeBoundingBox = Rect(
+                                    (barcodeBoundingBox.left * scaleX).toInt(),
+                                    (barcodeBoundingBox.top * scaleY).toInt(),
+                                    (barcodeBoundingBox.right * scaleX).toInt(),
+                                    (barcodeBoundingBox.bottom * scaleY).toInt()
+                            )
+
+                            // Check if the scaled bounding box is within the frame rectangle
+                            barcodeFrame?.frameRect!!.contains(scaledBarcodeBoundingBox)
```

Comment:
Avoid `!!` that would crash the app, prefer `?` that would gracefully do nothing

But `barcodeFrame` shouldn't be optional as we check for it at the beginning, you might be able to remove it
```suggestion
                            barcodeFrame.frameRect.contains(scaledBarcodeBoundingBox)
```